homeassistant/components/met_eireann/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""The met_eireann component."""
+from datetime import timedelta
+import logging
+from random import randrange
+
+import meteireann
+
+from homeassistant.const import (
+    CONF_ELEVATION,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    LENGTH_FEET,
+    LENGTH_METERS,
+)
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util.distance import convert as convert_distance
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured Met Éireann."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up Met Éireann as config entry."""
+    coordinator = MetEireannDataUpdateCoordinator(hass, config_entry)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
```

Comment:
Do you need the data to setup the entities in the platforms?  If not, then you can remove this check.